src/versionbits.cpp

```diff
@@ -267,3 +272,10 @@ void VersionBitsCache::Clear()
         caches[d].clear();
     }
 }
+
+AbstractEHFManager* AbstractEHFManager::globalInstance{nullptr};
+
+AbstractEHFManager* AbstractEHFManager::getInstance()
+{
+    return globalInstance;
+}
```

Comment:
I don't think I really like this new "AbstractEHFManager" thing; why not just have a global like we do now in other parts of codebase, or refactor it to not need a global and use member variables.

Reply:
it's to avoid circular dependency `versionbits -> mnhfmanager -> versionbits`.

This abstract interface is meant to break this circular dependency over a small simple interface `AbstractEHFManager`.

Reply:
If there is a logical dependency; then I think it's best for there to just be a circular dependency.

Reply:
there's nothing like "logical dependency". It just means that there's not enough interfaces between components; components should know public interfaces but not private details of implementatino.